subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/ClientForwardingProjectConfigurationOperationListener.java

```diff
@@ -60,4 +147,57 @@ private DefaultProjectConfigurationDescriptor toProjectConfigurationDescriptor(B
         return new DefaultProjectConfigurationDescriptor(id, displayName, parentId, details.getRootDir(), details.getProjectPath());
     }
 
+    private AbstractProjectConfigurationResult toProjectConfigurationOperationResult(OperationFinishEvent finishEvent, ProjectConfigurationResult configResult) {
+        long startTime = finishEvent.getStartTime();
+        long endTime = finishEvent.getEndTime();
+        Throwable failure = finishEvent.getFailure();
+        List<InternalPluginConfigurationResult> pluginConfigurationResults = configResult.toInternalPluginConfigurationResults();
+        if (failure != null) {
+            return new DefaultProjectConfigurationFailureResult(startTime, endTime, singletonList(DefaultFailure.fromThrowable(failure)), pluginConfigurationResults);
+        }
+        return new DefaultProjectConfigurationSuccessResult(startTime, endTime, pluginConfigurationResults);
+    }
+
+    private static class ProjectConfigurationResult {
+
+        private final Map<InternalPluginIdentifier, PluginConfigurationResult> pluginResults = new ConcurrentHashMap<>();
+
+        public void increment(InternalPluginIdentifier plugin, long applicationId, long duration) {
+            pluginResults.computeIfAbsent(plugin, key -> new PluginConfigurationResult(plugin, applicationId)).increment(duration);
+        }
+
+        public List<InternalPluginConfigurationResult> toInternalPluginConfigurationResults() {
+            return pluginResults.values().stream()
+                .sorted(comparing(PluginConfigurationResult::getFirstApplicationId))
```

Comment:
Why do we want to prescribe an ordering here? And if we do, wouldn't alphabetical make more sense?

Reply:
But what does that order really tell you, given that we aggregate events from subprojects as well?

Reply:
Fair enough. So you'd advocate for no particular order or alphabetical order?

Reply:
Probably alphabetical, in case someone dumps it straight to the screen.